servers/src/main/java/tachyon/master/file/meta/TTLBucketList.java

```diff
@@ -40,30 +40,53 @@ public TTLBucketList() {
   }
 
   /**
-   * Inserts an {@link InodeFile} to the appropriate bucket where its ttl end time lies in the
-   * bucket's interval, if no appropriate bucket exists, a new bucket will be created to contain
-   * this file, if ttl value is {@link Constants#NO_TTL}, the file won't be inserted to any buckets
-   * and nothing will happen.
+   * Gets the bucket in the list that contains the file.
    *
-   * @param file the file to be inserted
+   * @param file the file to be contained
+   * @return the bucket containing the file, or null if no such bucket exists
    */
-  public void insert(InodeFile file) {
+  private TTLBucket getBucketContaining(InodeFile file) {
     if (file.getTTL() == Constants.NO_TTL) {
-      return;
+      // no bucket will contain a file with NO_TTL.
+      return null;
     }
 
     long ttlEndTimeMs = file.getCreationTimeMs() + file.getTTL();
     // Gets the last bucket with interval start time less than or equal to the file's life end
     // time.
     TTLBucket bucket = mBucketList.floor(new TTLBucket(ttlEndTimeMs));
-    if (bucket == null || bucket.getTTLIntervalEndTimeMs() <= ttlEndTimeMs) {
+    if (bucket == null || bucket.getTTLIntervalEndTimeMs() < ttlEndTimeMs
+        || (bucket.getTTLIntervalEndTimeMs() == ttlEndTimeMs
+            && TTLBucket.getTTLIntervalMs() != 0)) {
       // 1. There is no bucket in the list, or
       // 2. All buckets' interval start time is larger than the file's life end time, or
       // 3. No bucket actually contains ttlEndTimeMs in its interval.
-      // So a new bucket should should be added with an appropriate interval start. Assume the list
-      // of buckets have continuous intervals, and the first interval starts at 0, then ttlEndTimeMs
-      // should be in number (ttlEndTimeMs / interval) interval, so the start time of this interval
-      // should be (ttlEndTimeMs / interval) * interval.
+      return null;
+    }
+
+    return bucket;
+  }
+
+  /**
+   * Inserts an {@link InodeFile} to the appropriate bucket where its ttl end time lies in the
+   * bucket's interval, if no appropriate bucket exists, a new bucket will be created to contain
+   * this file, if ttl value is {@link Constants#NO_TTL}, the file won't be inserted to any buckets
+   * and nothing will happen.
+   *
+   * @param file the file to be inserted
+   */
+  public void insert(InodeFile file) {
```

Comment:
Do we have a unit test for the TTLBucketList? If not, please add one and test the newly added functions (`insert` and `remove`).